File: sample-objc/ARGEARSample/UI/UIComponent/Main/Bottom/Filter/FilterCollectionViewCell.m
//
//  FilterCollectionViewCell.m
//  ARGEARSample
//
//  Created by Jihye on 10/09/2019.
//  Copyright © 2019 Seerslab. All rights reserved.
//

#import "FilterCollectionViewCell.h"

#define FILTER_CHECKED_FULLRATIO_COLOR [UIColor whiteColor]
#define FILTER_UNCHECKED_FULLRATIO_COLOR [UIColor colorWithRed:189.0f/255.0f green:189.0f/255.0f blue:189.0f/255.0f alpha:1.0f]
#define FILTER_CHECKED_COLOR [UIColor colorWithRed:97.0f/255.0f green:97.0f/255.0f blue:97.0f/255.0f alpha:1.0f]
#define FILTER_UNCHECKED_COLOR [UIColor colorWithRed:189.0f/255.0f green:189.0f/255.0f blue:189.0f/255.0f alpha:1.0f]

@implementation FilterCollectionViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)checked {
    [_filterCheckImage setHidden:NO];
}

- (void)unchecked {
    [_filterCheckImage setHidden:YES];
}

- (void)setData:(Item *)item {
    
    [_filterImage sd_setImageWithURL:[NSURL URLWithString:[item thumbnail]] placeholderImage:nil options:SDWebImageRefreshCached];
    [_filterNameLabel setText:[item title]];
}

- (void)setFilterNameLabelColor:(ARGMediaRatio)ratio {
    
    UIColor *textColor = [UIColor clearColor];
    UIColor *shadowColor = [UIColor clearColor];
    
    if (ratio == ARGMediaRatio_16x9) {
        
        textColor = [_filterCheckImage isHidden] ? FILTER_UNCHECKED_FULLRATIO_COLOR : FILTER_CHECKED_FULLRATIO_COLOR;
        shadowColor = [UIColor colorWithRed:97.f/255.f green:97.f/255.f blue:97.f/255.f alpha:1.0];
    } else {
        
        textColor = [_filterCheckImage isHidden] ? FILTER_UNCHECKED_COLOR : FILTER_CHECKED_COLOR;
    }
    
    [_filterNameLabel setTextColor:textColor];

    [[_filterNameLabel layer] setShadowOffset:CGSizeZero];
    [[_filterNameLabel layer] setShadowRadius:1.f];
    [[_filterNameLabel layer] setShadowOpacity:0.5f];
    [[_filterNameLabel layer] setMasksToBounds:NO];
    [[_filterNameLabel layer] setShouldRasterize:YES];
    [[_filterNameLabel layer] setShadowColor:[shadowColor CGColor]];
}

@end

File: sample-swift/ARGearSample/UI/UIComponent/Main/Bottom/Filter/FilterCollectionViewCell.swift
//
//  FilterCollectionViewCell.swift
//  ARGearSample
//
//  Created by Jihye on 2020/01/30.
//  Copyright © 2020 Seerslab. All rights reserved.
//

import UIKit
import ARGear
import SDWebImage

class FilterCollectionViewCell: UICollectionViewCell {
    
    let FILTER_CHECKED_FULLRATIO_COLOR = UIColor.white
    let FILTER_UNCHECKED_FULLRATIO_COLOR = UIColor(red: 189.0/255.0, green: 189.0/255.0, blue: 189.0/255.0, alpha: 1.0)
    let FILTER_CHECKED_COLOR = UIColor(red: 97.0/255.0, green: 97.0/255.0, blue: 97.0/255.0, alpha: 1.0)
    let FILTER_UNCHECKED_COLOR = UIColor(red: 189.0/255.0, green: 189.0/255.0, blue: 189.0/255.0, alpha: 1.0)
    
    @IBOutlet weak var filterImage: UIImageView!
    @IBOutlet weak var filterCheckImage: UIImageView!
    @IBOutlet weak var filterNameLabel: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    func setRatio(_ ratio: ARGMediaRatio) {
        
        var textColor = UIColor.clear
        var shadowColor = UIColor.clear
        
        if ratio == ._16x9 {
            textColor = filterCheckImage.isHidden ? FILTER_UNCHECKED_FULLRATIO_COLOR : FILTER_CHECKED_FULLRATIO_COLOR
            shadowColor = UIColor(red: 97.0/255.0, green: 97.0/255.0, blue: 97.0/255.0, alpha: 1.0)
        } else {
            textColor = filterCheckImage.isHidden ? FILTER_UNCHECKED_COLOR : FILTER_CHECKED_COLOR
        }
        
        self.filterNameLabel.textColor = textColor
        
        self.filterNameLabel.layer.shadowOffset = CGSize.zero
        self.filterNameLabel.layer.shadowRadius = 1.0
        self.filterNameLabel.layer.shadowOpacity = 0.5
        self.filterNameLabel.layer.masksToBounds = false
        self.filterNameLabel.layer.shouldRasterize = true
        self.filterNameLabel.layer.shadowColor = shadowColor.cgColor
    }
    
    func checked() {
        self.filterCheckImage.isHidden = false
        filterNameLabel.isHidden = false
    }
    
    func unchecked() {
        self.filterCheckImage.isHidden = true
        filterNameLabel.isHidden = false
    }
    
    func disableButtonCell(ratio: ARGMediaRatio) {
        filterCheckImage.isHidden = true
        filterNameLabel.isHidden = true
        
        var iconName = "icDisabledBlackOff"
        if ratio == ._16x9 {
            iconName = "icDisabledWhiteOff"
        }
        
        filterImage.image = UIImage(named: iconName)
    }
    
    func setFilter(_ filter: Item) {
        guard
            let title = filter.title,
            let thumbnailUrl = filter.thumbnail
            else {
                return
        }
        
        self.filterNameLabel.text = title
        self.filterImage.sd_setImage(with: URL(string: thumbnailUrl), placeholderImage: nil, options: [.refreshCached], completed: nil)
    }
}
